### DISCUSSION TITLE
Better documentation of built-ins

### DISCUSSION THREAD

**Discussion by ktk** [eyereasoner/eye#67] [General]
IMO the documentation of the [Notation3 built-ins](https://w3c.github.io/N3/files/builtins.html) is not very good for various reasons. One of it is the lack of any styling but the more important one is the lack of examples.

The SPARQL 1.1 spec is much better in documenting the functions, see for example some of the string functions: https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#func-strlen

I'm happy to contribute for the simple stuff but I think it should be discussed how and where.

**Comment by phochste** [comment]
Yes, the W3C Notation3 group is currently working on a rewriting the documentation with examples for each built-in. In the mean-time, I've started in the eye reasoner implementation repository an [Notation3-By-Example](https://github.com/eyereasoner/Notation3-By-Example) project containing many examples with in line documentation.
